kedro-datasets/kedro_datasets/partitions/partitioned_dataset.py

```diff
@@ -28,6 +28,18 @@
 S3_PROTOCOLS = ("s3", "s3a", "s3n")
 
 
+def _grandparent(path: str) -> str:
```

Comment:
It would be helpful to have a docstring here explaining what the method is doing.